### ISSUE TITLE
Harbor 2.0.0 login problem with XSRF token

### ISSUE THREAD

**Issue by hasonhai** [goharbor/harbor#11935]
I use the helm chart 1.3.2 to deploy the stable version of Harbor 2.0.0 on Kubernetes.
Everything look fine as expected. But I cannot login both from the browser and from the docker commandline with the admin user. When I disabled the XSRF, I can log in from the browser, but still not success with docker. 

Below is my config for the "core" component.
```
  app.conf: |
    appname = Harbor
    runmode = dev
    enablegzip = true

    [dev]
    httpport = 8080
    EnableXSRF = false
    XSRFKey = n6PThFSMBpobyc6lDLbQvOCyiarC9gDAGypEAGsX
    XSRFExpire = 3600
 
```
When login from Docker:

```
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id 30d4d05fc6a0e3ac3f21e637cae7d81a to the logger for the request GET /v2/
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/security/unauthorized.go:29][requestID="30d4d05fc6a0e3ac3f21e637cae7d81a"]: an unauthorized security context generated for request GET /v2/
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:53]: In artifact info middleware, url: /v2/
2020-05-14T12:47:17Z [DEBUG] [/server/error/error.go:58]: {"errors":[{"code":"UNAUTHORIZED","message":"unauthorized: unauthorized"}]}
2020/05/14 12:47:17.600 [D] [middleware.go:52]  |      127.0.0.1| 401 |   3.473658ms|   match| GET      /v2/   r:/v2/*
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id e07b5c742783342ed562abf64840701f to the logger for the request GET /service/token
2020-05-14T12:47:17Z [ERROR] [/server/middleware/security/oidc_cli.go:54][requestID="e07b5c742783342ed562abf64840701f"]: failed to verify secret: user is not onboarded as OIDC user
2020-05-14T12:47:17Z [DEBUG] [/common/dao/user.go:277]: Check if user admin is super user
2020-05-14T12:47:17Z [DEBUG] [/core/auth/authenticator.go:139]: Current AUTH_MODE is db_auth
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/security/basic_auth.go:48][requestID="e07b5c742783342ed562abf64840701f"]: a basic auth security context generated for request GET /service/token
2020-05-14T12:47:17Z [DEBUG] [/core/service/token/token.go:35]: URL for token request: /service/token?account=admin&client_id=docker&offline_token=true&service=harbor-registry
2020-05-14T12:47:17Z [DEBUG] [/core/service/token/creator.go:203]: scopes: []
2020-05-14T12:47:17Z [DEBUG] [/core/service/token/authutils.go:48]: scopes: []
2020/05/14 12:47:17.645 [D] [middleware.go:52]  |      127.0.0.1| 200 |   5.315389ms|   match| GET      /service/token   r:/service/token
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id fe3cc734f59d0bd4590dbcf368d4ec86 to the logger for the request GET /v2/
2020-05-14T12:47:17Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:53]: In artifact info middleware, url: /v2/
2020/05/14 12:47:17.675 [D] [middleware.go:52]  |      127.0.0.1| 401 |   10.18987ms|   match| GET      /v2/   r:/v2/*
```

**Expected behavior and actual behavior:**
I also notice that, the normal health check command also reponse with error.

Below is the health check in kubernetes:

```
        livenessProbe:
          httpGet:
            path: /api/v2.0/ping
            port: 8080
          initialDelaySeconds: {{ .Values.core.livenessProbe.initialDelaySeconds }}
          periodSeconds: 10
        readinessProbe:
          httpGet:
            path: /api/v2.0/ping
            port: 8080
          initialDelaySeconds: 20
          periodSeconds: 10

```
and the error logs:
```
2020-05-14T09:27:11Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id bec00d05-8556-4b15-9e6e-fab199fe0ffe to the logger for the request GET /api/v2.0/ping
2020-05-14T09:27:11Z [DEBUG] [/server/middleware/security/unauthorized.go:29][requestID="bec00d05-8556-4b15-9e6e-fab199fe0ffe"]: an unauthorized security context generated for request GET /api/v2.0/ping
2020/05/14 09:27:11.671 [D] [transaction.go:62]  |  10.145.229.68| 200 |   2.994348ms|   match| GET      /api/v2.0/ping   r:/api/v2.0/ping
```
and when oidc is configured:
```
2020-05-14T12:29:41Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id f8415a61-cf67-42c3-8484-ab492debf521 to the logger for the request GET /api/v2.0/ping
2020-05-14T12:29:41Z [DEBUG] [/common/utils/oidc/helper.go:193]: Raw ID token for verification:
2020-05-14T12:29:41Z [WARNING] [/server/middleware/security/idtoken.go:45][requestID="f8415a61-cf67-42c3-8484-ab492debf521"]: failed to verify token: oidc: malformed jwt: square/go-jose: compact JWS format must have three parts
2020-05-14T12:29:41Z [DEBUG] [/server/middleware/security/unauthorized.go:29][requestID="f8415a61-cf67-42c3-8484-ab492debf521"]: an unauthorized security context generated for request GET /api/v2.0/ping
2020/05/14 12:29:41.675 [D] [transaction.go:62]  |  10.145.229.68| 200 |   2.697543ms|   match| GET      /api/v2.0/ping   r:/api/v2.0/ping
```

**Steps to reproduce the problem:**
Install Harbor 2.0,0 with helm chart 1.3.2, modify the health check for "core" from /api/ping to /api/v2.0/ping.

**Versions:**
- harbor version: [2.0.0]
- docker engine version: [19.03.8]
- kubernetes version: [1.16.8]

**Comment by angelbarrera92** [comment]
I've faced the same error. It is working if you disable xsrf

Try:
```
  app.conf: |
    appname = Harbor
    runmode = dev
    enablegzip = true

    [dev]
    httpport = 8080
```

**Comment by hasonhai** [comment]
@angelbarrera92 yes it works for most of the features. But when login by docker CLI, I still got code 401 for Unauthorized. I saw in the logs, after disabled XSRF, docker is authorized at the "core", but it got rejected at the harbor-registry component because of no or not-valid authorizing context. I guess the "core" failed to create a valid token to talk to harbor-registry. I wonder if it's related XSRF?

```
time="2020-05-14T14:22:48.360020473Z" level=debug msg="authorizing request" go.version=go1.13.8 http.request.host=harborv2.xxx.com http.request.id=9cb91e75-948a-4027-bf46-22ad09a41517 http.request.method=GET http.request.remoteaddr=127.0.0.1 http.request.uri="/v2/" http.request.useragent="docker/19.03.8 go/go1.12.17 git-commit/afacb8b kernel/3.10.0-1127.8.2.el7.x86_64 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.8 \(linux\))"
10.42.1.31 - - [14/May/2020:14:22:48 +0000] "GET /v2/ HTTP/1.1" 401 87 "" "docker/19.03.8 go/go1.12.17 git-commit/afacb8b kernel/3.10.0-1127.8.2.el7.x86_64 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.8 \\(linux\\))"
time="2020-05-14T14:22:48.360105474Z" level=warning msg="error authorizing context: authorization token required" go.version=go1.13.8 http.request.host=harborv2.xxx.com http.request.id=9cb91e75-948a-4027-bf46-22ad09a41517 http.request.method=GET http.request.remoteaddr=127.0.0.1 http.request.uri="/v2/" http.request.useragent="docker/19.03.8 go/go1.12.17 git-commit/afacb8b kernel/3.10.0-1127.8.2.el7.x86_64 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.8 \(linux\))"
```
In addition, I also tried disabling redirecturl in the registry. The problem still persisted.

**Comment by reasonerjt** [comment]
@hasonhai 
The cm in the helm chart does not work with 2.0.0 build.

It is being reworked to work with 2.0.0 images, there will be a new release of the helm chart in the next couple of days.

**Comment by bitsf** [comment]
waiting @heww merging PR for helm

**Comment by heww** [comment]
Closed as harbor-helm for 2.0.0 released.

**Comment by aceeric** [comment]
I also am experiencing this with Helm version 2.4.1 and nginx as an Ingress.
